state/protocol/protocol_state/epochs/statemachine_test.go

```diff
@@ -515,3 +515,46 @@ func (s *EpochStateMachineSuite) TestEvolveState_EventsAreFiltered() {
 	})
 	require.NoError(s.T(), err)
 }
+
+// TestEvolveStateTransitionToNextEpoch_WithInvalidStateTransition tests that EpochStateMachine transitions to the next epoch
+// if an invalid state transition has been detected in a block which triggers transition to the next epoch.
+// In such situation we still need to enter the next epoch but with the fallback state machine.
```

Comment:
I think we do _not explicitly_ test that the `fallback state machine` is used here, but rather that the output satisfies certain requirements. That the implementation used the fallback state machine under the hood could be viewed as an implementation detail in my opinion. 
```suggestion
// In such situation, we still need to enter the next epoch (because it has already been committed), but persist in the 
// state that we have entered Epoch fallback mode (`flow.ProtocolStateEntry.InvalidEpochTransitionAttempted` is set to `true`).
```